pyogrio/geopandas.py

```diff
@@ -411,11 +415,17 @@ def write_dataframe(
         else:
             crs = geometry.crs.to_wkt(WktVersion.WKT1_GDAL)
 
+    # Prepare geometry array to write
+    if geometry_column is not None:
+        geometry = to_wkb(geometry.values)
+    else:
+        geometry = np.repeat(None, len(df))
```

Comment:
I could be wrong (didn't test the idea), but it seems like with some relatively minor changes in `_io.pyx::ogr_write(...)` that it would be possible to pass in `geometry=None` and avoid passing in an array of `None` values.

We may also want to raise an error if `geometry` is `None` and there is no field data.

For instance, writing a data frame that has no columns produces different behavior depending on the driver: GDAL automatically creates a `FID` column if there are no columns passed in, whereas geopackage will be read back as an empty data frame (by default; it _does_ create an FID column as well)

Reply:
I had given it a brief thought, but because the number of rows was based on the geometry field and it is possible not to have any columns I didn't pursue it.

However, having our discussion in #263 about reading no geometry nor any columns in mind, as well as your following suggestion it should indeed not be a problem to do so.
    `We may also want to raise an error if geometry is None and there is no field data.`

So, I just committed a version that supports this (and fixes support for field_data and fields being None as well).